lib/class-wp-theme-json-gutenberg.php

```diff
@@ -65,6 +65,7 @@ class WP_Theme_JSON_Gutenberg {
 			'padding' => null,
 		),
 		'typography' => array(
+			'columnCount'    => null,
```

Comment:
Can we change the attribute to be `textColumns` instead?

Reply:
Well, technically, it's not _just_ columns for text: something using this property could have images and other media in its content.

In fact, are we even sure this belongs in the category of typography tools? It technically has use cases outside of bare text, especially once you start using `break-inside` to preserve the integrity of child containers. A gallery could hypothetically use it to layout its images, for example, as could a series of listing of social media posts. I'd argue it actually belongs with the layout tools.

I think "content columns" is a more strictly accurate term than "text columns".

Reply:
Well, the original issues are about unlocking a specific effect on textual content. I'm not sure we should design it as a general tool and expose it as one were it'd start to compete with other layout tools. "Content columns" is just very close to regular columns, which are also about showing any content in columns. Here we need to represent the flow and continuity aspect, which seems better captured by "text columns" to me. It might be better placed in "layout" regardless, but with the right context.

There's also a question on which blocks this should be exposed. For example, wouldn't make any sense to have this on the headings block.

Reply:
The interesting thing about CSS `columns` versus typical divs-in-divs "columns" is that:

- Divs-in-divs columns have content explicitly assigned to a column, and the height of each column can vary wildly.
- CSS `columns` have content dynamically placed in the columns, often attempting to balance the height of each column (though a minimum content height for a column can be set).

These have big ramifications for when you would want to use one over the other, but I have no clue how to convey these subtle but important differences to users.

---

Here's the only blocks I can think of where this control would make sense:

- Buttons
- Gallery
- Group
- List
- Navigation
- Paragraph
- Quote
- Verse

In other words: most of the container blocks, and _maybe_ a few standard text blocks for the sake of allowing users to avoid extra divs in a few cases. In most cases, though, you'd actually want to avoid using CSS `columns` on a single Paragraph, since that's less future-proof for adding additional text. So maybe it should only be the container blocks that have the control, to encourage best practices.